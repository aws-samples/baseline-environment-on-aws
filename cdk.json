{
  "app": "npx ts-node bin/able-base-sa.ts",
  "context": {
    "dev": {
      "description": "Environment variables for able-guest-*-samples.ts",
      "envName": "Development",
      "vpcCidr": "10.100.0.0/16",
      "securityNotifyEmail": "notify-security@example.com",
      "monitoringNotifyEmail": "notify-monitoring@example.com",
      "dbUser": "dbadmin",
      "slackNotifier": {
        "workspaceId": "T8XXXXXXX",
        "channelIdSec": "C01XXXXXXXX",
        "channelIdMon": "C01YYYYYYYY"
      }
    },
    "stage": {
      "description": "Environment variables for able-guest-*-samples.ts",
      "envName": "Staging",
      "vpcCidr": "10.110.0.0/16",
      "securityNotifyEmail": "notify-security@example.com",
      "monitoringNotifyEmail": "notify-monitoring@example.com",
      "dbUser": "dbadmin",
      "slackNotifier": {
        "workspaceId": "T8XXXXXXX",
        "channelIdSec": "C01XXXXXXXX",
        "channelIdMon": "C01YYYYYYYY"
      }
    },
    "prod": {
      "description": "Environment variables for able-guest-*-samples.ts",
      "envName": "Production",
      "vpcCidr": "10.120.0.0/16",
      "securityNotifyEmail": "notify-security@example.com",
      "monitoringNotifyEmail": "notify-monitoring@example.com",
      "dbUser": "dbadmin",
      "slackNotifier": {
        "workspaceId": "T8XXXXXXX",
        "channelIdSec": "C01XXXXXXXX",
        "channelIdMon": "C01YYYYYYYY"
      }
    },
    "ctaudit": {
      "description": "Environment variables for able-base-ct-audit.ts",
      "env": {
        "account": "123456789012",
        "region": "ap-northeast-1"
      },
      "slackNotifier": {
        "workspaceId": "T8XXXXXXX",
        "channelIdAgg": "C01ZZZZZZZZ"
      }
    }
  }
}
