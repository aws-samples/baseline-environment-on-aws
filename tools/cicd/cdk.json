{
  "app": "npx ts-node --prefer-ts-exts bin/bleadeploy.ts",
  "context": {
    "dev": {
      "description": "Context samples for Dev - Anonymous account & region",
      "envName": "Development",
      "githubRepositoryOwner": "ownername",
      "githubRepositoryName": "repositoryname",
      "githubTargetBranch": "staging",
      "codestarConnectionArn": "arn:aws:codestar-connections:ap-northeast-1:xxxxxxxxxxxx:connection/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
    },
    "stage": {
      "description": "Context samples for Staging - Specific account & region  ",
      "env": {
        "account": "111111111111",
        "region": "ap-northeast-1"
      },
      "envName": "Staging",
      "githubRepositoryOwner": "ownername",
      "githubRepositoryName": "repositoryname",
      "githubTargetBranch": "staging",
      "codestarConnectionArn": "arn:aws:codestar-connections:ap-northeast-1:xxxxxxxxxxxx:connection/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
    },
    "prod": {
      "description": "Context samples for Production - Specific account & region  ",
      "env": {
        "account": "222222222222",
        "region": "ap-northeast-1"
      },
      "envName": "Production",
      "githubRepositoryOwner": "ownername",
      "githubRepositoryName": "repositoryname",
      "githubTargetBranch": "main",
      "codestarConnectionArn": "arn:aws:codestar-connections:ap-northeast-1:xxxxxxxxxxxx:connection/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
    }
  }
}
