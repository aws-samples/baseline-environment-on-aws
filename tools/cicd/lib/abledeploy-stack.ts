import * as cdk from "@aws-cdk/core";
import * as iam from "@aws-cdk/aws-iam";
import * as codebuild from "@aws-cdk/aws-codebuild";

export interface ABLEdeployStackProps extends cdk.StackProps {
  githubRepositoryOwner: string;
  githubRepositoryName: string;
  githubTargetBranch: string;
}

export class ABLEdeployStack extends cdk.Stack {
  constructor(scope: cdk.Construct, id: string, props: ABLEdeployStackProps) {
    super(scope, id, props);

    // Create GitHub source for CodeBuild
    const gitHubSource = codebuild.Source.gitHub({
      owner: props.githubRepositoryOwner,
      repo: props.githubRepositoryName,
      webhook: true,
      fetchSubmodules: true,
      webhookFilters: [
        codebuild.FilterGroup.inEventOf(codebuild.EventAction.PUSH).andBranchIs(
          props.githubTargetBranch
        ),
      ],
    });

    // Create CodeBuild from GitHub source
    const project = new codebuild.Project(this, "ABLEdeployProject", {
      source: gitHubSource,
    });

    // Change role for CodeBuild to attache Admin access for using CDK
    const role = project.role;
    if (role) {
      role.addManagedPolicy(
        iam.ManagedPolicy.fromAwsManagedPolicyName("AdministratorAccess")
      );
    }
  }
}
