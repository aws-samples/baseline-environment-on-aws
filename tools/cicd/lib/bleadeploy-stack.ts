import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { aws_iam as iam } from 'aws-cdk-lib';
import { aws_codebuild as codebuild } from 'aws-cdk-lib';
import { aws_codepipeline as codepipeline } from 'aws-cdk-lib';
import { aws_codepipeline_actions as codepipeline_actions } from 'aws-cdk-lib';

export interface BLEAdeployStackProps extends cdk.StackProps {
  githubRepositoryOwner: string;
  githubRepositoryName: string;
  githubTargetBranch: string;
  codestarConnectionArn: string;
}

export class BLEAdeployStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props: BLEAdeployStackProps) {
    super(scope, id, props);

    const pipeline = new codepipeline.Pipeline(this, `${id}-Pipeline`, {
      pipelineName: `${id}-Pipeline`,
    });

    // You just have to select GitHub as the source when creating the connection in the console
    const sourceOutput = new codepipeline.Artifact();
    const sourceAction = new codepipeline_actions.CodeStarConnectionsSourceAction({
      actionName: `${id}-GitHubSource`,
      owner: props.githubRepositoryOwner,
      repo: props.githubRepositoryName,
      branch: props.githubTargetBranch,
      connectionArn: props.codestarConnectionArn,
      output: sourceOutput,
    });

    const deployRole = new iam.Role(this, `${id}-CodeBuildDeployRole`, {
      assumedBy: new iam.ServicePrincipal('codebuild.amazonaws.com'),
      managedPolicies: [
        {
          managedPolicyArn: 'arn:aws:iam::aws:policy/AdministratorAccess',
        },
      ],
    });

    const applicationBuild = new codebuild.PipelineProject(this, `${id}-CodeBuildProject`, {
      projectName: `${id}-CodeBuildProject`,
      role: deployRole,
      environment: {
        buildImage: codebuild.LinuxBuildImage.AMAZON_LINUX_2_3,
        environmentVariables: {
          AWS_DEFAULT_REGION: {
            type: codebuild.BuildEnvironmentVariableType.PLAINTEXT,
            value: this.region,
          },
        },
      },
    });

    const buildAction = new codepipeline_actions.CodeBuildAction({
      actionName: `${id}-BuildAction`,
      project: applicationBuild,
      input: sourceOutput,
      runOrder: 3,
    });

    pipeline.addStage({
      stageName: 'Source',
      actions: [sourceAction],
    });
    pipeline.addStage({
      stageName: 'Build',
      actions: [buildAction],
    });
  }
}
