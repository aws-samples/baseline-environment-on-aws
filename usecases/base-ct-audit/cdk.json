{
  "app": "npx ts-node --prefer-ts-exts bin/blea-base-ct-audit.ts",
  "context": {
    "dev-audit": {
      "description": "Context samples for ControlTower Audit Account - Specific account & region",
      "env": {
        "account": "333333333333",
        "region": "ap-northeast-1"
      },
      "slackNotifier": {
        "workspaceId": "T8XXXXXXX",
        "channelIdAgg": "C01ZZZZZZZZ"
      }
    }
  }
}
