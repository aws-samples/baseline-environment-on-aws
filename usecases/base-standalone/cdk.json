{
  "app": "npx ts-node --prefer-ts-exts bin/blea-base-sa.ts",
  "context": {
    "dev": {
      "description": "Context samples for Dev - Anonymous account & region",
      "envName": "Development",
      "securityNotifyEmail": "notify-security@example.com",
      "slackNotifier": {
        "workspaceId": "T8XXXXXXX",
        "channelIdSec": "C00XXXXXXXX"
      }
    },
    "stage": {
      "description": "Context samples for Staging - Specific account & region  ",
      "env": {
        "account": "111111111111",
        "region": "ap-northeast-1"
      },
      "envName": "Staging",
      "securityNotifyEmail": "notify-security@example.com",
      "slackNotifier": {
        "workspaceId": "T8XXXXXXX",
        "channelIdSec": "C01XXXXXXXX"
      }
    },
    "prod": {
      "description": "Context samples for Production - Specific account & region  ",
      "env": {
        "account": "222222222222",
        "region": "ap-northeast-1"
      },
      "envName": "Production",
      "securityNotifyEmail": "notify-security@example.com",
      "slackNotifier": {
        "workspaceId": "T8XXXXXXX",
        "channelIdSec": "C02XXXXXXXX"
      }
    }
  }
}
