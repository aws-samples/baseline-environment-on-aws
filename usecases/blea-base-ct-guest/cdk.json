{
  "app": "npx ts-node --prefer-ts-exts bin/blea-base-ct-guest.ts"
}
