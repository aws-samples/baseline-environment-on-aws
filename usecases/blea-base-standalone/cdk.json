{
  "app": "npx ts-node --prefer-ts-exts bin/blea-base-standalone.ts"
}
