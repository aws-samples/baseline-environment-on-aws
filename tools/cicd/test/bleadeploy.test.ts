import { Template } from 'aws-cdk-lib/assertions';
import * as cdk from 'aws-cdk-lib';
import * as BLEAdeploy from '../lib/bleadeploy-stack';

// Load cdk.json to get context parameters
import * as cdk_json from '../cdk.json';

const procEnv = {
  account: process.env.CDK_DEFAULT_ACCOUNT,
  region: process.env.CDK_DEFAULT_REGION,
};

const pjPrefix = 'BLEA';
const app = new cdk.App();
const envKey = 'dev';
const envVals = cdk_json['context'][envKey];

describe(`${pjPrefix} Stacks`, () => {
  test('BLEA Deploy Stack', () => {
    const deployStack = new BLEAdeploy.BLEAdeployStack(app, 'DeployStack', {
      githubRepositoryOwner: envVals['githubRepositoryOwner'],
      githubRepositoryName: envVals['githubRepositoryName'],
      githubTargetBranch: envVals['githubTargetBranch'],
      codestarConnectionArn: envVals['codestarConnectionArn'],
      env: procEnv,
    });

    // test with snapshot
    expect(Template.fromStack(deployStack)).toMatchSnapshot();
  });
});
