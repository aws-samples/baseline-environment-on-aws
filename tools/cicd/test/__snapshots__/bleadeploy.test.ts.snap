// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BLEA Stacks BLEA Deploy Stack 1`] = `
Object {
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "DeployStackCodeBuildDeployRoleAAA60B15": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "codebuild.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          "arn:aws:iam::aws:policy/AdministratorAccess",
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "DeployStackCodeBuildDeployRoleDefaultPolicyFA3548DF": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "logs:CreateLogGroup",
                "logs:CreateLogStream",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:",
                      Object {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":log-group:/aws/codebuild/",
                      Object {
                        "Ref": "DeployStackCodeBuildProject9086D16F",
                      },
                    ],
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:",
                      Object {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":log-group:/aws/codebuild/",
                      Object {
                        "Ref": "DeployStackCodeBuildProject9086D16F",
                      },
                      ":*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "codebuild:CreateReportGroup",
                "codebuild:CreateReport",
                "codebuild:UpdateReport",
                "codebuild:BatchPutTestCases",
                "codebuild:BatchPutCodeCoverages",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":codebuild:",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":report-group/",
                    Object {
                      "Ref": "DeployStackCodeBuildProject9086D16F",
                    },
                    "-*",
                  ],
                ],
              },
            },
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "DeployStackPipelineArtifactsBucket2FBBF0B9",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "DeployStackPipelineArtifactsBucket2FBBF0B9",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "kms:Decrypt",
                "kms:DescribeKey",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "DeployStackPipelineArtifactsBucketEncryptionKeyD623D0F7",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": Array [
                "kms:Decrypt",
                "kms:Encrypt",
                "kms:ReEncrypt*",
                "kms:GenerateDataKey*",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "DeployStackPipelineArtifactsBucketEncryptionKeyD623D0F7",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "DeployStackCodeBuildDeployRoleDefaultPolicyFA3548DF",
        "Roles": Array [
          Object {
            "Ref": "DeployStackCodeBuildDeployRoleAAA60B15",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "DeployStackCodeBuildProject9086D16F": Object {
      "Properties": Object {
        "Artifacts": Object {
          "Type": "CODEPIPELINE",
        },
        "Cache": Object {
          "Type": "NO_CACHE",
        },
        "EncryptionKey": Object {
          "Fn::GetAtt": Array [
            "DeployStackPipelineArtifactsBucketEncryptionKeyD623D0F7",
            "Arn",
          ],
        },
        "Environment": Object {
          "ComputeType": "BUILD_GENERAL1_SMALL",
          "EnvironmentVariables": Array [
            Object {
              "Name": "AWS_DEFAULT_REGION",
              "Type": "PLAINTEXT",
              "Value": Object {
                "Ref": "AWS::Region",
              },
            },
          ],
          "Image": "aws/codebuild/amazonlinux2-x86_64-standard:3.0",
          "ImagePullCredentialsType": "CODEBUILD",
          "PrivilegedMode": false,
          "Type": "LINUX_CONTAINER",
        },
        "Name": "DeployStack-CodeBuildProject",
        "ServiceRole": Object {
          "Fn::GetAtt": Array [
            "DeployStackCodeBuildDeployRoleAAA60B15",
            "Arn",
          ],
        },
        "Source": Object {
          "Type": "CODEPIPELINE",
        },
      },
      "Type": "AWS::CodeBuild::Project",
    },
    "DeployStackPipelineArtifactsBucket2FBBF0B9": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "BucketEncryption": Object {
          "ServerSideEncryptionConfiguration": Array [
            Object {
              "ServerSideEncryptionByDefault": Object {
                "KMSMasterKeyID": Object {
                  "Fn::GetAtt": Array [
                    "DeployStackPipelineArtifactsBucketEncryptionKeyD623D0F7",
                    "Arn",
                  ],
                },
                "SSEAlgorithm": "aws:kms",
              },
            },
          ],
        },
        "PublicAccessBlockConfiguration": Object {
          "BlockPublicAcls": true,
          "BlockPublicPolicy": true,
          "IgnorePublicAcls": true,
          "RestrictPublicBuckets": true,
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
    "DeployStackPipelineArtifactsBucketEncryptionKeyAlias11AD99EB": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "AliasName": "alias/codepipeline-deploystackdeploystackpipeline17211b4e",
        "TargetKeyId": Object {
          "Fn::GetAtt": Array [
            "DeployStackPipelineArtifactsBucketEncryptionKeyD623D0F7",
            "Arn",
          ],
        },
      },
      "Type": "AWS::KMS::Alias",
      "UpdateReplacePolicy": "Delete",
    },
    "DeployStackPipelineArtifactsBucketEncryptionKeyD623D0F7": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "KeyPolicy": Object {
          "Statement": Array [
            Object {
              "Action": "kms:*",
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::KMS::Key",
      "UpdateReplacePolicy": "Delete",
    },
    "DeployStackPipelineArtifactsBucketPolicy9C58EA6A": Object {
      "Properties": Object {
        "Bucket": Object {
          "Ref": "DeployStackPipelineArtifactsBucket2FBBF0B9",
        },
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "s3:*",
              "Condition": Object {
                "Bool": Object {
                  "aws:SecureTransport": "false",
                },
              },
              "Effect": "Deny",
              "Principal": Object {
                "AWS": "*",
              },
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "DeployStackPipelineArtifactsBucket2FBBF0B9",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "DeployStackPipelineArtifactsBucket2FBBF0B9",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
    "DeployStackPipelineBuildDeployStackBuildActionCodePipelineActionRoleDB81A0F3": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "DeployStackPipelineBuildDeployStackBuildActionCodePipelineActionRoleDefaultPolicy888F32DF": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "codebuild:BatchGetBuilds",
                "codebuild:StartBuild",
                "codebuild:StopBuild",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "DeployStackCodeBuildProject9086D16F",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "DeployStackPipelineBuildDeployStackBuildActionCodePipelineActionRoleDefaultPolicy888F32DF",
        "Roles": Array [
          Object {
            "Ref": "DeployStackPipelineBuildDeployStackBuildActionCodePipelineActionRoleDB81A0F3",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "DeployStackPipelineE72A6049": Object {
      "DependsOn": Array [
        "DeployStackPipelineRoleDefaultPolicyFFF0FC4A",
        "DeployStackPipelineRole9A958AB1",
      ],
      "Properties": Object {
        "ArtifactStore": Object {
          "EncryptionKey": Object {
            "Id": Object {
              "Fn::GetAtt": Array [
                "DeployStackPipelineArtifactsBucketEncryptionKeyD623D0F7",
                "Arn",
              ],
            },
            "Type": "KMS",
          },
          "Location": Object {
            "Ref": "DeployStackPipelineArtifactsBucket2FBBF0B9",
          },
          "Type": "S3",
        },
        "Name": "DeployStack-Pipeline",
        "RoleArn": Object {
          "Fn::GetAtt": Array [
            "DeployStackPipelineRole9A958AB1",
            "Arn",
          ],
        },
        "Stages": Array [
          Object {
            "Actions": Array [
              Object {
                "ActionTypeId": Object {
                  "Category": "Source",
                  "Owner": "AWS",
                  "Provider": "CodeStarSourceConnection",
                  "Version": "1",
                },
                "Configuration": Object {
                  "BranchName": "staging",
                  "ConnectionArn": "arn:aws:codestar-connections:ap-northeast-1:xxxxxxxxxxxx:connection/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
                  "FullRepositoryId": "ownername/repositoryname",
                },
                "Name": "DeployStack-GitHubSource",
                "OutputArtifacts": Array [
                  Object {
                    "Name": "Artifact_Source_DeployStack-GitHubSource",
                  },
                ],
                "RoleArn": Object {
                  "Fn::GetAtt": Array [
                    "DeployStackPipelineSourceDeployStackGitHubSourceCodePipelineActionRoleF0A57F33",
                    "Arn",
                  ],
                },
                "RunOrder": 1,
              },
            ],
            "Name": "Source",
          },
          Object {
            "Actions": Array [
              Object {
                "ActionTypeId": Object {
                  "Category": "Build",
                  "Owner": "AWS",
                  "Provider": "CodeBuild",
                  "Version": "1",
                },
                "Configuration": Object {
                  "ProjectName": Object {
                    "Ref": "DeployStackCodeBuildProject9086D16F",
                  },
                },
                "InputArtifacts": Array [
                  Object {
                    "Name": "Artifact_Source_DeployStack-GitHubSource",
                  },
                ],
                "Name": "DeployStack-BuildAction",
                "RoleArn": Object {
                  "Fn::GetAtt": Array [
                    "DeployStackPipelineBuildDeployStackBuildActionCodePipelineActionRoleDB81A0F3",
                    "Arn",
                  ],
                },
                "RunOrder": 3,
              },
            ],
            "Name": "Build",
          },
        ],
      },
      "Type": "AWS::CodePipeline::Pipeline",
    },
    "DeployStackPipelineRole9A958AB1": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "codepipeline.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "DeployStackPipelineRoleDefaultPolicyFFF0FC4A": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "DeployStackPipelineArtifactsBucket2FBBF0B9",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "DeployStackPipelineArtifactsBucket2FBBF0B9",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "kms:Decrypt",
                "kms:DescribeKey",
                "kms:Encrypt",
                "kms:ReEncrypt*",
                "kms:GenerateDataKey*",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "DeployStackPipelineArtifactsBucketEncryptionKeyD623D0F7",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "DeployStackPipelineSourceDeployStackGitHubSourceCodePipelineActionRoleF0A57F33",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "DeployStackPipelineBuildDeployStackBuildActionCodePipelineActionRoleDB81A0F3",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "DeployStackPipelineRoleDefaultPolicyFFF0FC4A",
        "Roles": Array [
          Object {
            "Ref": "DeployStackPipelineRole9A958AB1",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "DeployStackPipelineSourceDeployStackGitHubSourceCodePipelineActionRoleDefaultPolicyACEE3BD8": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "codestar-connections:UseConnection",
              "Effect": "Allow",
              "Resource": "arn:aws:codestar-connections:ap-northeast-1:xxxxxxxxxxxx:connection/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
            },
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "DeployStackPipelineArtifactsBucket2FBBF0B9",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "DeployStackPipelineArtifactsBucket2FBBF0B9",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "kms:Decrypt",
                "kms:DescribeKey",
                "kms:Encrypt",
                "kms:ReEncrypt*",
                "kms:GenerateDataKey*",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "DeployStackPipelineArtifactsBucketEncryptionKeyD623D0F7",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": "s3:PutObjectAcl",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    Object {
                      "Fn::GetAtt": Array [
                        "DeployStackPipelineArtifactsBucket2FBBF0B9",
                        "Arn",
                      ],
                    },
                    "/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "DeployStackPipelineSourceDeployStackGitHubSourceCodePipelineActionRoleDefaultPolicyACEE3BD8",
        "Roles": Array [
          Object {
            "Ref": "DeployStackPipelineSourceDeployStackGitHubSourceCodePipelineActionRoleF0A57F33",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "DeployStackPipelineSourceDeployStackGitHubSourceCodePipelineActionRoleF0A57F33": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
